File: src/main/java/com/addressbook/model/Person.java
package com.addressbook.model;

import lombok.*;
import org.apache.ignite.cache.query.annotations.QuerySqlField;

import java.io.Serializable;
import java.util.UUID;

@Getter
@Setter
@AllArgsConstructor
@ToString
public class Person implements Serializable {

    @QuerySqlField(index = true)
    private String id;

    @QuerySqlField(index = true)
    private String orgId;

    @QuerySqlField(index = true)
    private String firstName;

    @QuerySqlField(index = true)
    private String lastName;

    @QuerySqlField(index = true)
    private String resume;

    @QuerySqlField(index = true)
    private String salary;

    public Person() {
        id = UUID.randomUUID().toString();
    }

    public Person(Organization org, String firstName, String lastName, String salary, String resume) {
        this.id = UUID.randomUUID().toString();
        this.orgId = org.getId();
        this.firstName = firstName;
        this.lastName = lastName;
        this.salary = salary;
        this.resume = resume;
    }

    public Person(String id, String firstName, String lastName) {
        this.id = id;
        this.firstName = firstName;
        this.lastName = lastName;
    }
}

File: addressbook-common/src/main/kotlin/com/addressbook/model/Person.kt
package com.addressbook.model

import dev.morphia.annotations.Entity
import dev.morphia.annotations.Id
import org.apache.ignite.cache.query.annotations.QuerySqlField
import java.io.Serializable
import java.util.*

@Entity("persons")
@javax.persistence.Entity
class Person : Serializable {

    @Id
    @javax.persistence.Id
    @QuerySqlField(index = true)
    var id: String? = null

    @QuerySqlField(index = true)
    var orgId: String? = null

    @QuerySqlField(index = true)
    var firstName: String? = null

    @QuerySqlField(index = true)
    var lastName: String? = null

    @QuerySqlField(index = true)
    var resume: String? = null

    @QuerySqlField(index = true)
    var salary: String? = null

    constructor(org: Organization, firstName: String, lastName: String, salary: String, resume: String) : this() {
        this.orgId = org.id
        this.firstName = firstName
        this.lastName = lastName
        this.salary = salary
        this.resume = resume
    }

    constructor() {
        id = UUID.randomUUID().toString()
    }

    constructor(id: String, firstName: String, lastName: String) : this() {
        this.id = id
        this.firstName = firstName
        this.lastName = lastName
    }
}
